### PR TITLE
SSCS-10784 Panel Type & Panel Specialism schema change

### PR DESCRIPTION
**Pull request by JustMehmet**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10784

### Change description ###

HearingsPanelMapping.java: Updated method getRoleTypes at line 56, added benefitCode parameter to getRoleTypes method at line 48.
ServiceHearingValuesMapping.java: Added benefitCode parameter to getRoleTypes method at line 48.
HearingsPanelMappingTest.java: Added 2 new tests which checks getRoleTypes method according to given benefitCode.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java [hunk 1]
```diff
@@ -100,6 +112,10 @@ public static List<String> getPanelSpecialisms(@Valid SscsCaseData caseData, Ses
         if (isNull(sessionCategoryMap)) {
             return panelSpecialisms;
         }
+        // if benefit is child support specialism should be empty
+        if (isNotBlank(caseData.getBenefitCode()) && caseData.getBenefitCode().equals(CHILD_SUPPORT.getBenefitCode())) {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java:116]: What about IIDB ?

**Review by ss-evoco — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java (hunk 1 above)
Comment (JustMehmet) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java:116]: In the task for IIDB it is asked to put all the selected panel specialisms. That is why I did not control IIDB. ( https://tools.hmcts.net/jira/browse/SSCS-10784 )

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java [hunk 2]
```diff
@@ -40,17 +45,24 @@ private HearingsPanelMapping() {
...
             .panelSpecialisms(getPanelSpecialisms(caseData, getSessionCaseCode(caseData, referenceDataServiceHolder)))
             .build();
     }
 
-    public static List<String> getRoleTypes() {
-        //TODO Need to retrieve RoleTypes from caseData and/or ReferenceData
-        return Collections.emptyList();
+    public static List<String> getRoleTypes(String benefitCode) {
+        List<String> roleTypes = new ArrayList<>();
+        if (isNotBlank(benefitCode)) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java:58]: I don't think we should be implementing things like this. If we're asked to add more cases there will be more and more if blocks. I'd recommend something like creating an enum that we can search based on the benefit code. This enum would also have a consumer which takes in List<String> (roletypes). You'd find the corresponding enum then invoke its consumer which should add the correct reference to the list.

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java (hunk 2 above)
Comment (JustMehmet) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java]: Added a new ENUM as BenefitRoleRelationType and retrieved role types by using it' s static method as below:

public static List<String> getRoleTypes(String benefitCode) {
        return BenefitRoleRelationType.findRoleTypesByBenefitCode(benefitCode);
    }

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by ikirsanov — APPROVED** [review]

**Pull request merged**